components/proxy/src/main/java/com/hotels/styx/proxy/ProxyConnectorFactory.java

```diff
@@ -149,6 +150,7 @@ public void configure(Channel channel, HttpHandler httpPipeline) {
                     .addLast("connection-throttler", excessConnectionRejector)
                     .addLast("idle-handler", new IdleStateHandler(serverConfig.requestTimeoutMillis(), 0, serverConfig.keepAliveTimeoutMillis(), MILLISECONDS))
                     .addLast("channel-stats", channelStatsHandler)
+                    .addLast("channel-connection-stats", new ConnectionCountHandler(metrics))
```

Comment:
This is not entirely true - the `channelActivated` field needs to be in a non-shared handler, but the channel count needs to be shared. However, it is true that the implementation can be simplified and I will do so.

(Mikko is no longer reviewing this PR, but I am leaving this comment so that other reviewers understand that I have taken the words into account.)